presto-orc/src/main/java/com/facebook/presto/orc/writer/ColumnWriterUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.writer;
+
+import com.facebook.presto.orc.checkpoint.BooleanStreamCheckpoint;
+import com.facebook.presto.orc.checkpoint.StreamCheckpoint;
+import com.facebook.presto.orc.metadata.RowGroupIndex;
+import com.facebook.presto.orc.metadata.statistics.ColumnStatistics;
+import com.facebook.presto.orc.stream.PresentOutputStream;
+import com.facebook.presto.orc.stream.ValueOutputStream;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ColumnWriterUtils
+{
+    private ColumnWriterUtils()
+    {
+        // utils
+    }
+
+    /**
+     * Build RowGroupIndex using column statistics and checkpoints.
+     */
+    @SafeVarargs
+    public static List<RowGroupIndex> buildRowGroupIndexes(
+            boolean compressed,
+            List<ColumnStatistics> rowGroupColumnStatistics,
+            Optional<List<? extends StreamCheckpoint>> prependCheckpoints,
+            PresentOutputStream presentStream,
+            ValueOutputStream<? extends StreamCheckpoint>... dataStreams)
+    {
+        ImmutableList.Builder<RowGroupIndex> rowGroupIndexes = ImmutableList.builder();
```

Comment:
Should we validate all checkpoints has the same length as rowGroupStatistics ?

Reply:
Thoughts on using List<List<Checkpoint>> instead of varargs here.

I thought, it would enforce at least one data stream, but Struct field will not have any checkpoints.